From c91f78f6d87b8f9f7cd948973c5add12348a2d51 Mon Sep 17 00:00:00 2001
From: Cheng Jian <cj.chengjian@huawei.com>
Date: Thu, 11 Nov 2021 16:28:49 +0800
Subject: [PATCH] disable stealing by default
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c91f78f6d87b8f9f7cd948973c5add12348a2d51


hulk inclusion
category: feature
bugzilla: 38261, https://gitee.com/openeuler/kernel/issues/I49XPZ
CVE: NA

---------------------------

Steal tasks to improve CPU utilization can solve some performance
problems such as mysql, but not all scenarios are optimized, such as
hackbench.

So turn off by default.

Signed-off-by: Cheng Jian <cj.chengjian@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/features.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index 98925e29b1b9..d7c63040deb7 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -60,7 +60,7 @@ SCHED_FEAT(SIS_PROP, true)
  * Steal a CFS task from another CPU when going idle.
  * Improves CPU utilization.
  */
-SCHED_FEAT(STEAL, true)
+SCHED_FEAT(STEAL, false)
 
 /*
  * Issue a WARN when we do multiple update_rq_clock() calls
-- 
2.26.2

